google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/stub/ErrorCountPerConnectionTest.java

```diff
@@ -0,0 +1,263 @@
+      if (i % 3 == 0) {
+        query = Query.create(ERROR_TABLE_NAME);
+        errorCount += 1;
+      } else {
+        query = Query.create(SUCCESS_TABLE_NAME);
+      }
+      try {
+        stub.readRowsCallable().call(query).iterator().hasNext();
+      } catch (Exception e) {
+        // noop
+      }
+    }
+    ArgumentCaptor<Long> errorCountCaptor = ArgumentCaptor.forClass(long.class);
+    Mockito.doNothing()
+        .when(statsRecorderWrapperForConnection)
+        .putAndRecordPerConnectionErrorCount(errorCountCaptor.capture());
+    runInterceptorTasksAndAssertCount(1);
+    List<Long> allErrorCounts = errorCountCaptor.getAllValues();
+    assertThat(allErrorCounts.size()).isEqualTo(1);
+    assertThat(allErrorCounts.get(0)).isEqualTo(errorCount);
+
+    errorCount = 0;
+
+    for (int i = 0; i < 20; i++) {
+      Query query;
+      if (i % 3 == 0) {
+        query = Query.create(SUCCESS_TABLE_NAME);
+      } else {
+        query = Query.create(ERROR_TABLE_NAME);
+        errorCount += 1;
+      }
+      try {
+        stub.readRowsCallable().call(query).iterator().hasNext();
+      } catch (Exception e) {
+        // noop
+      }
+    }
+    errorCountCaptor = ArgumentCaptor.forClass(long.class);
+    Mockito.doNothing()
+        .when(statsRecorderWrapperForConnection)
+        .putAndRecordPerConnectionErrorCount(errorCountCaptor.capture());
+    runInterceptorTasksAndAssertCount(1);
+    allErrorCounts = errorCountCaptor.getAllValues();
+    assertThat(allErrorCounts.size()).isEqualTo(1);
+    assertThat(allErrorCounts.get(0)).isEqualTo(errorCount);
+  }
+
+  @Test
+  public void ignoreInactiveConnection() throws Exception {
+    EnhancedBigtableStub.create(builder.build());
+
+    ArgumentCaptor<Long> errorCountCaptor = ArgumentCaptor.forClass(long.class);
+    Mockito.doNothing()
+        .when(statsRecorderWrapperForConnection)
+        .putAndRecordPerConnectionErrorCount(errorCountCaptor.capture());
+    runInterceptorTasksAndAssertCount(1);
+    List<Long> allErrorCounts = errorCountCaptor.getAllValues();
+    assertThat(allErrorCounts).isEmpty();
+  }
+
+  @Test
+  public void noFailedRequests() throws Exception {
+    EnhancedBigtableStub stub = EnhancedBigtableStub.create(builder.build());
+
+    for (int i = 0; i < 20; i++) {
+      try {
+        stub.readRowsCallable().call(Query.create(SUCCESS_TABLE_NAME)).iterator().hasNext();
+      } catch (Exception e) {
+        // noop
+      }
+    }
+    ArgumentCaptor<Long> errorCountCaptor = ArgumentCaptor.forClass(long.class);
+    Mockito.doNothing()
+        .when(statsRecorderWrapperForConnection)
+        .putAndRecordPerConnectionErrorCount(errorCountCaptor.capture());
+    runInterceptorTasksAndAssertCount(1);
+    List<Long> allErrorCounts = errorCountCaptor.getAllValues();
+    assertThat(allErrorCounts.size()).isEqualTo(1);
+    assertThat(allErrorCounts.get(0)).isEqualTo(0);
+  }
+
+  private void runInterceptorTasksAndAssertCount(int expectNumOfTasks) {
```

Comment:
now we're only creating 1 stub for all the test, is this step still necessary?

Reply:
I cleaned up the function to remove the argument, now we just check to make sure the number of tasks created is always 1.